queries/sourcepawn/highlights.scm

```diff
@@ -0,0 +1,306 @@
+
+; Types
+[
+  (builtin_type)
+  (any_type)
+] @type.builtin
+
+(type
+  (identifier) @type)
+
+; Variables
+(variable_declaration
+  name: (identifier) @variable)
+
+(old_variable_declaration
+  name: (identifier) @variable)
+
+; Preprocessor
+[
+  (preproc_include)
+  (preproc_tryinclude)
+] @keyword.import
+
+[
+  (system_lib_string)
+  (string_literal)
+] @string
+
+(preproc_arg) @constant
+
+(preproc_macro) @function.macro
+
+(macro_param) @variable.parameter
+
+[
+  (preproc_assert)
+  (preproc_pragma)
+  (preproc_if)
+  (preproc_else)
+  (preproc_elseif)
+  (preproc_endif)
+  (preproc_endinput)
+  (preproc_error)
+  (preproc_warning)
+] @keyword.directive
+
+[
+  (preproc_define)
+  (preproc_undefine)
+] @keyword.directive.define
+
+(preproc_define
+  name: (identifier) @constant)
+
+(preproc_undefine
+  name: (identifier) @constant)
+
+; Expressions
+(view_as) @function.builtin
+
+(sizeof_expression) @function.macro
+
+[
+  (this)
+  ; https://github.com/alliedmodders/sourcemod/blob/5c0ae11a4619e9cba93478683c7737253ea93ba6/plugins/include/handles.inc#L78
+  (hardcoded_symbol)
+] @variable.builtin
+
+; Comments
+(comment) @comment
+
+; General
+(parameter_declaration
+  defaultValue: (identifier) @constant)
+
+[
+  (fixed_dimension)
+  (dimension)
+  (array_indexed_access)
+] @punctuation.bracket
+
+(escape_sequence) @string.escape
+
+; Constructors
+(new_expression
+  class: (identifier) @type
+  arguments: (call_arguments) @constructor)
+
+; Methodmaps
+(methodmap) @type.definition
+
+(methodmap
+  name: (identifier) @type)
+
+(methodmap
+  inherits: (identifier) @type)
+
+(methodmap_method_constructor
+  name: (identifier) @constructor)
+
+(methodmap_method
+  name: (identifier) @function.method)
+
+(methodmap_native
+  name: (identifier) @function.method)
+
+(methodmap_property
+  name: (identifier) @property)
+
+[
+  (methodmap_property_getter)
+  (methodmap_property_setter)
+] @function.method
+
+; Enum structs
+(enum_struct) @type.definition
+
+(enum_struct
+  name: (identifier) @type)
+
+(enum_struct_field
+  name: (identifier) @variable.member)
+
+(enum_struct_method
+  name: (identifier) @function.method)
+
+; Non-type Keywords
+[
+  (variable_storage_class)
+  (visibility)
+  (variable_storage_class)
+] @keyword.storage
+
+(assertion) @function.builtin
+
+(function_declaration_kind) @keyword.function
+
+[
+  "new"
+  "delete"
+] @keyword.operator
+
+[
+  "."
+  ","
+] @punctuation.delimiter
+
+; Operators
+[
+  "+"
+  "-"
+  "*"
+  "/"
+  "%"
+  "++"
+  "--"
+  "="
+  "+="
+  "-="
+  "*="
+  "/="
+  "=="
+  "!="
+  "<"
+  ">"
+  ">="
+  "<="
+  "!"
+  "&&"
+  "||"
+  "&"
+  "|"
+  "~"
+  "^"
+  "<<"
+  ">>"
+  ">>>"
+  "|="
+  "&="
+  "^="
+  "~="
+  "<<="
+  ">>="
+  (ignore_argument)
+  (scope_access)
+  (rest_operator)
+] @operator
+
+; public Plugin myinfo
+(struct_declaration
+  name: (identifier) @variable.builtin)
+
+; Typedef/Typedef
+[
+  (typedef)
+  (functag)
+  (funcenum)
+  (typeset)
+] @type.definition
```

Comment:
Note that `@type.definition` is meant to be used for the type alias and not the keyword that defines it.

Reply:
Ah in that case, this was incorrect, it has been removed now.